src/core/services.py

```diff
@@ -508,3 +511,73 @@ def queue_job(project_id: int, job_id: int, session: Session) -> None:
         logger.error(
             f"Cannot queue job {job_id}, it's of status {job.status()}."
         )
+
+
+def get_directory_listing(path: str) -> Dict[str, Any]:
+    """Get contents found in a given directory. Does not search recursivly.
+
+    Parameters
+    ----------
+    path    :   str
+        Path to the folder one wants to get listing from.
+
+    Raises
+    ------
+    NotADirectoryError
+        When the given path is a file and not a directory.
+
+    FileNotFoundError
+        Directory cannot be found at a given path.
+
+    Returns
+    -------
+    Dict
+        jsTree formatted json containing information about root node at path.
+    """
+    root_node: Dict[str, Any] = dict()
+    child_list = []
+
+    if isfile(path):
+        raise NotADirectoryError("Path must be a directory.")
+
+    if not isdir(path):
+        raise FileNotFoundError("Directory at '{}' was not found.".format(path))
+
+    root, dirs, files = next(os.walk(path))
+
+    # Ensure alphabetical order
+    dirs.sort()
+    files.sort()
+
+    # Append directories
+    child_list.extend(
+        [
+            {
+                "text": name,
+                "type": "folder",
+            }
+            for name in dirs
+        ]
+    )
+
+    # Append files
+    child_list.extend(
+        [
+            {
+                "text": name,
+                "type": str(guess_type(name)[0])
+                if guess_type(name)[0]
+                else "file",
+            }
+            for name in files
+        ]
+    )
+
+    # Create root node
+    root_node = {
+        "text": root,
+        "type": "folder",
```

Comment:
```suggestion
        "type": "root",
```

Could maybe return a list of root nodes.  If multiple folders are set to be mounted.  Internal folder, external drive, etc.

Reply:
Not sure if this is a good idea, if you request a second time in a subfolder, the selected folder will now be a "root"

Reply:
I would serve this when no parameter is provided.  Then it will always be true for root.

Reply:
Ok, that can be arranged. Will work on this tomorrow.